### PR TITLE
Adding initial configuration for testing redhat-developer/odo on prow

### PR DESCRIPTION
**Pull request by mohammedzee1000**
This PR adds initial configurations for running redhat-developer/odo tests on OpenShift Prow
Ref : https://github.com/redhat-developer/odo/issues/1167

### CODE REVIEW

File: ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    #      - destination_dir: .
+    #        source_path: /usr/bin/oc
+    #  to: your-output-image
+
+tests:
+- artifact_dir: /tmp/artifacts
+  as: unit
+  commands: make test
+  container:
+    from: golang
```
Comment (stevekuznetsov) [inline ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml]: I think you want `from: src` here

File: ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    name: release
+    namespace: openshift
+    tag: golang-1.10
+
+base_images:
+  cli:
+    name: '4.0'
+    namespace: ocp
+    tag: cli
+  golang:
```
Comment (stevekuznetsov) [inline ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml]: You don't need this -- these are base images for output image builds, you get golang as a build root as configured above

File: cluster/ci/config/prow/plugins.yaml [hunk 3]
```diff
@@ -496,6 +519,7 @@ triggers:
   - kiegroup/kie-cloud-operator
   - containers/libpod
   - operator-framework
+  - redhat-developer/odo
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/plugins.yaml]: Remove this one -- you're in the next stanza already. This stanza will make it required to be in the `OpenShift` org to trigger jobs

File: ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml (hunk 1 above)
Comment (mohammedzee1000) [reply ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml]: Sure

File: ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml (hunk 2 above)
Comment (mohammedzee1000) [reply ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml]: ack thanks

File: cluster/ci/config/prow/plugins.yaml (hunk 3 above)
Comment (mohammedzee1000) [reply cluster/ci/config/prow/plugins.yaml]: Done

**Comment by mohammedzee1000** [comment]
Rerun after https://github.com/redhat-developer/odo/pull/1412

**Comment by mohammedzee1000** [comment]
Sorry for butchering your username @bbguimaraes

**Pull request merged**